### ISSUE TITLE
Issue with MTASDocumentIndexing tasks interaction

### ISSUE THREAD

**Issue by reckart** [inception-project/inception#2252]
**Describe the bug**
A clear and concise description of what the bug is. Maybe related to recent switch to the `SchedulerService` in the `SearchService`. Maybe some kind of race-condition.

**To Reproduce**
Unit test seems to fail spuriously: `de.tudarmstadt.ukp.inception.search.index.mtas.MtasDocumentIndexTest.testLimitQueryToDocument`

**Expected behavior**
Should never fail.

**Screenshots**
```
Expecting:
  [SearchResult[documentId=2,documentTitle=Raw text document 1,tokenStart=1,tokenLength=1,offsetStart=4,offsetEnd=11,leftContext=The ,text=capital,rightContext= of Galicia is],
    SearchResult[documentId=3,documentTitle=Raw text document 2,tokenStart=1,tokenLength=1,offsetStart=4,offsetEnd=11,leftContext=The ,text=capital,rightContext= of Portugal is],
    SearchResult[documentId=3,documentTitle=Raw text document 2,tokenStart=1,tokenLength=1,offsetStart=4,offsetEnd=11,leftContext=The ,text=capital,rightContext= of Portugal is]]
to contain exactly in any order:
  [SearchResult[documentId=2,documentTitle=Raw text document 1,tokenStart=1,tokenLength=1,offsetStart=4,offsetEnd=11,leftContext=The ,text=capital,rightContext= of Galicia is],
    SearchResult[documentId=3,documentTitle=Raw text document 2,tokenStart=1,tokenLength=1,offsetStart=4,offsetEnd=11,leftContext=The ,text=capital,rightContext= of Portugal is]]
but the following elements were unexpected:
  [SearchResult[documentId=3,documentTitle=Raw text document 2,tokenStart=1,tokenLength=1,offsetStart=4,offsetEnd=11,leftContext=The ,text=capital,rightContext= of Portugal is]]
when comparing values using field/property by field/property comparator on all fields/properties
Comparators used:
- for elements fields (by type): {Double -> DoubleComparator[precision=1.0E-15], Float -> FloatComparator[precision=1.0E-6]}
- for elements (by type): {Double -> DoubleComparator[precision=1.0E-15], Float -> FloatComparator[precision=1.0E-6]}
	at de.tudarmstadt.ukp.inception.search.index.mtas.MtasDocumentIndexTest.testLimitQueryToDocument(MtasDocumentIndexTest.java:392)
Standard Output
=== testLimitQueryToDocument === testLimitQueryToDocument()=====================
2021-05-02 18:40:32 [main] INFO MtasDocumentIndexTest - Uploading document via documentService.uploadSourceDocument: ([Raw text document 1](null),The capital of Galicia is Santiago de Compostela.)
2021-05-02 18:40:32 [main] INFO DocumentServiceImpl - Imported source document [Raw text document 1](2) to project [TestLimitQueryToDocument](2)
2021-05-02 18:40:32 [main] TRACE SearchServiceImpl - Starting afterDocumentCreate
2021-05-02 18:40:32 [main] INFO MtasDocumentIndexTest - Uploading document via documentService.uploadSourceDocument: ([Raw text document 2](null),The capital of Portugal is Lissabon.)
2021-05-02 18:40:32 [recommender-worker-4] TRACE SearchServiceImpl - Loading index for project [TestLimitQueryToDocument](2)
2021-05-02 18:40:32 [recommender-worker-4] TRACE SearchServiceImpl - Initializing persistent index state in project [TestLimitQueryToDocument](2).
2021-05-02 18:40:32 [recommender-worker-4] DEBUG SearchServiceImpl - Request to index source document [Raw text document 1](2) in project [TestLimitQueryToDocument](2)
2021-05-02 18:40:32 [recommender-worker-4] TRACE SearchServiceImpl - Index in project [TestLimitQueryToDocument](2) has not yet been initialized. Scheduling an asynchronous re-indexing.
2021-05-02 18:40:32 [recommender-worker-1] INFO SearchServiceImpl - Re-indexing project [TestLimitQueryToDocument](2) 
2021-05-02 18:40:32 [main] INFO DocumentServiceImpl - Imported source document [Raw text document 2](3) to project [TestLimitQueryToDocument](2)
2021-05-02 18:40:32 [main] TRACE SearchServiceImpl - Starting afterDocumentCreate
2021-05-02 18:40:32 [main] INFO MtasDocumentIndexTest - Waiting for uploaded documents to be indexed...
2021-05-02 18:40:32 [recommender-worker-2] DEBUG SearchServiceImpl - Request to index source document [Raw text document 2](3) in project [TestLimitQueryToDocument](2)
2021-05-02 18:40:32 [recommender-worker-1] DEBUG SearchServiceImpl - Request to index source document [Raw text document 1](2) in project [TestLimitQueryToDocument](2)
2021-05-02 18:40:32 [recommender-worker-1] DEBUG SearchServiceImpl - Request to index source document [Raw text document 2](3) in project [TestLimitQueryToDocument](2)
2021-05-02 18:40:37 [main] INFO MtasDocumentIndexTest - Indexing complete!
2021-05-02 18:40:37 [main] TRACE SearchServiceImpl - Query [capital] for user [admin] in project [TestLimitQueryToDocument](2)
2021-05-02 18:40:37 [main] TRACE SearchServiceImpl - Query [capital] for user [admin] in project [TestLimitQueryToDocument](2)
```

**Please complete the following information:**
 - Version and build ID: 0.20.0-SNAPSHOT

**Comment by reckart** [comment]
I believe the problem is that a re-indexing is triggered and in parallel normal document indexing tasks are triggered. If a re-indexing is already running, then document indexing tasks for the project should be deferred until the re-indexing is complete. Also, if a re-indexing is triggered, and queued but not started indexing tasks for the project should be discarded and the re-indexing should wait until any already running indexing tasks are complete.
